cs3api4lab/auth/authenticator.py

```diff
@@ -1,31 +1,47 @@
-import cs3.gateway.v1beta1.gateway_api_pb2 as cs3gw
-import time
-import cs3.gateway.v1beta1.gateway_api_pb2_grpc as cs3gw_grpc
+import importlib
+from datetime import datetime
+
+import jwt
+
 from cs3api4lab.config.config_manager import Cs3ConfigManager
-from cs3api4lab.auth.channel_connector import ChannelConnector
 
 
 class Authenticator:
-    tokens = {}  # map client_id [string] to {authentication token, token expiration time}
+    instance = None
+    config = None
+
+    def __init__(self, config=None, log=None):
 
-    def __init__(self, config, channel):
+        if config is None:
+            config = Cs3ConfigManager().get_config()
         self.config = config
-        self.cs3_api = cs3gw_grpc.GatewayAPIStub(channel)
+
+        if self.instance is None:
+
+            if log is not None:
+                log.info(f"Authenticate with method {self.config['authenticator_class']}")
+
+            class_name = self.config['authenticator_class'].split('.')[-1]
+            module_name = self.config['authenticator_class'].split(class_name)[0]
+            module_name = module_name.rstrip('.')
+
+            if class_name == "Authenticator":
+                raise AttributeError("Can't instance Authenticator class")
+
+            module = importlib.import_module(module_name)
+            clazz = getattr(module, class_name)
+            self.instance = clazz(config=self.config, log=log)
```

Comment:
At this point, I'm using a different way to create instance a class because It is generating from the config value. I didn't remember, but I tried to using typical instance class from JupyterLab (trait type). We have parent classes with basic methods. Each children classes must implement the unique code.